### DISCUSSION TITLE
Contributions are Welcome!

### DISCUSSION THREAD

**Discussion by EvgSkv** [EvgSkv/logica#30] [General]
We are welcoming contributions to the Logica project!

If you want to get involved there is a lot you can do, for example:

* **Use Logica!** If you want to use Logica for your business project, or explore it by solving some fun problems we'd be interested in getting feedback and answering questions that you have on the way.
* **Contribute to the code!** Logica is reasonably mature and can be applied to real world problems. Yet there is a lot more that can be done. Full support for PosgreSQL and SQLite is needed. More engines should be onboarded. We are happy to discuss ideas of further development that you have.
* **Bring your thoughts, spread the word!** Discuss Logica with your friends and colleagues, bring new insights and perspectives to this forum.

**Comment by odanoburu** [comment]
Hi @EvgSkv , thanks for the cool project!

I'm looking to start contributing, but I'm stuck on how to do a local editable installation, as I'm used to do with `pip install -e .` (where `.` is the logica root directory). pip complains there's no setup.py file.

How should I do it?

**Reply by EvgSkv** [reply]
@odanoburu thanks for your interest!
Suggested flow is that when developing you test your changes by simply run logica from the cloned repository folder. Like
```
./logica - run Greet <<<'@Engine("sqlite"); Greet("Hello!")'
```
or
```
python3 logica.py - run Greet <<<'@Engine("sqlite"); Greet("Hello!")'
```

This way your scripts that are supposed to use production logica install are unaffected and you don't have to use environments.

Let me know if you have further questions and please let me know if you see major issues with this approach.

**Reply by odanoburu** [reply]
Oh great! It does work fine, thank you!
